dashboard/app/models/user.rb

```diff
@@ -619,9 +619,14 @@ def user_level_locked?(script_level, level)
   # script that hasn't yet been passed, starting its search at the last level we submitted
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
+    user_levels = user_levels_by_level.values.flatten.select do |user_level|
```

Comment:
It seems simpler to do `user_levels.where(script: script)` than `user_levels_by_level(script).values.flatten` if you want to simplify pre-existing code. Aside, I don't understand why the `flatten` is necessary.